src/DependencyRewriterV2.php

```diff
@@ -58,6 +67,28 @@ public function __construct(?callable $applicationFactory = null, $composerFile
         };
     }
 
+    public function onPreCommandRun(PreCommandRunEvent $event)
+    {
+        parent::onPreCommandRun($event);
+        $input = $event->getInput();
+
+        $options = [];
+        foreach (self::COMPOSER_LOCK_UPDATE_OPTIONS as $optionName) {
```

Comment:
Reminder to myself:
Find another way to extract passed arguments to `composer` as this package may not be installed when command is being executed.

So there should be away of getting these arguments during runtime and not by hooking into this event.

Reply:
I don't understand your note to yourself. It sounds like you're worried about identifying those arguments when the plugin is not yet installed... are you thinking about a scenario where the plugin is listed in the application, but dependencies have never been installed, so Composer installs the plugins _first_, and then continues installing other dependencies?

In that case, doesn't Composer essentially do a second pass?

If it doesn't... you could always look at `$argv`, as it's a superglobal, and neither Composer nor symfony/console change the values it contains.

Reply:
> are you thinking about a scenario where the plugin is listed in the application, but dependencies have never been installed, so Composer installs the plugins first, and then continues installing other dependencies?

Yes, that was one of those scenarios I stumbled on when implementing composer v2 support.

> In that case, doesn't Composer essentially do a second pass?

No it doesn't. Only if the plugin was available when executing the command, the plugin is being called.
So yes, I was thinking about that `$argv` aswell or probably some OOP way. Need to verify the possibilities.